### PR TITLE
usbh: add experimental host support

### LINKED ISSUE #12386: Support USB Host on Zephyr
**Is your feature request related to a problem? Please describe.**
We're working for UNISOC and doing upstream for new chip and board.
https://github.com/zephyrproject-rtos/zephyr/pull/10626
Now we need to support pen drive when Zephyr is working on USB host mode.
but there is USB device mode only on Zephyr.

**Describe the solution you'd like**
Please provide a framework of USB host, like driver, api, protocol, etc..

**Describe alternatives you've considered**
Please provide thinking and methods that I might use to add.

**Additional context**
NA

### PR DESCRIPTION
**Pull request by efdev**
This PR is experimental USB host support for Zephyr. 
The work was done during a student internship at PHYTEC Messtechnik GmbH.
The work is based on uC/USBH Host implementation and currently supported classes are the hub and msc class.  

Fixes: #12386

### CODE REVIEW

File: samples/subsys/usbh/README.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+Overview
+********
+
+A simple sample to use an USB host controller to write and read a block
+on a mass storage device.
+
+Requirements
+************
+
+This sample has been tested with with the atsamr21_xpro and an USB hub with
```
Comment (aep) [inline samples/subsys/usbh/README.rst:15]: typo

**Comment by bob2oneil** [comment]
Is there any estimate on when and if this pull request will make into into the standard main branch?

**Comment by coran21** [comment]
Any update on this? Would be really great to have USB host support.

**Comment by avigreen1978** [comment]
Any updates on this?

**Pull request closed without merging**